### PR TITLE
POC to improve project test readability

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

/kind cleanup
/kind feature
/kind code-refactoring

**What does this PR do / why we need it**:
This pr improves readability for project scenario for now.

**Which issue(s) this PR fixes**:

Fixes - part of #4741 

**PR acceptance criteria**:

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

- [ ] Update changelog

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

`make test-cmd-project` should run successfully

### CODE REVIEW

File: tests/integration/project/project_tests_test.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package project
+
+import (
+	. "github.com/onsi/ginkgo"
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo project command tests", func() {
+
+	var commonVar helper.CommonVar
```
Comment (prietyc123) [inline tests/integration/project/project_tests_test.go:10]: Failure is causing due to `commonVar` is not accessible across the project package. 
I will fix it soon.

**Comment by prietyc123** [comment]
Works fine locally against 4.7 OCP cluster

```
$ make test-cmd-project
ginkgo  -randomizeAllSpecs -slowSpecThreshold=120 -timeout 14400s -nodes=1 -focus="odo project command tests" tests/integration/project/
Running Suite: Project Suite
============================
Random Seed: 1621961885 - Will randomize all specs
Will run 7 of 7 specs
•••••••
JUnit report was created: /Users/pkumari/go/src/github.com/openshift/odo/tests/reports/junit_2021-5-25_22-28-09_1.xml
Ran 7 of 7 Specs in 57.460 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
Ginkgo ran 1 suite in 1m1.817234012s
Test Suite Passed
```

**Comment by prietyc123** [comment]
It would be great if somebody share some quick thought on the linter failure
```
INFO[2021-05-26T04:25:00Z] level=warning msg="[runner] Can't run linter unused: buildir: analysis skipped: errors in package: [/go/src/github.com/openshift/odo/tests/integration/project/project_validator.go:32:46: undeclared name: commonVar /go/src/github.com/openshift/odo/tests/integration/project/project_validator.go:55:35: undeclared name: commonVar /go/src/github.com/openshift/odo/tests/integration/project/project_validator.go:60:112: undeclared name: commonVar /go/src/github.com/openshift/odo/tests/integration/project/project_validator.go:60:152: undeclared name: commonVar]"
```

**Comment by prietyc123** [comment]
kubernetes tests are failing due to https://github.com/openshift/odo/issues/4733

**Comment by prietyc123** [comment]
Due to 
```
++ oc apply -n openshift -f https://raw.githubusercontent.com/openshift/library/master/arch/x86_64/community/wildfly/imagestreams/wildfly-centos7.json 
INFO[2021-06-02T10:06:36Z] error: unable to read URL "https://raw.githubusercontent.com/openshift/library/master/arch/x86_64/community/wildfly/imagestreams/wildfly-centos7.json", server reported 500 Internal Server Error, status code=500
```
Wildfly component was not recognised and 4.7 test failed.

/test v4.7-integration-e2e

File: tests/integration/project/project_test_scenario.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	. "github.com/onsi/ginkgo"
+)
+
+func ProjectTestScenario() {
+	When("Executing Machine readable output tests", func() {
+		It("Help for odo project list should contain machine output", HelpCommand)
+		It("should be able to get project", GetProject)
+		It("should display the help", HelpCommandProject)
+		It("should display only the project name", CommandWithQFlag)
```
Comment (dharmit) [inline tests/integration/project/project_test_scenario.go:12]: This one has nothing to do with machine-readable output either. It should be in a different spec, IMO.

File: tests/integration/project/project_test_scenario.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+		It("--wait should work with deleting a project", CommandWaitFlag)
```
Comment (dharmit) [inline tests/integration/project/project_test_scenario.go:13]: This one has nothing to do with machine-readable output either. It should be in a different spec, IMO.

File: tests/integration/project/project_validator.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	helper.CmdShouldPass("odo", "project", "create", projectName)
+
+	// Delete with --wait
+	output := helper.CmdShouldPass("odo", "project", "delete", projectName, "-f", "--wait")
+	Expect(output).To(ContainSubstring("Waiting for project to be deleted"))
+}
+
+func CommandDeleteWithJsonFlag() {
+	projectName := helper.RandString(6)
+	helper.CmdShouldPass("odo", "project", "create", projectName, "-o", "json")
```
Comment (dharmit) [inline tests/integration/project/project_validator.go:47]: Why are we creating the project with json flag here? We don't seem to be testing anything specific to json output for project create.

**Comment by kadel** [comment]
Now when I see this implemented I'm even more worried about this approach than before.

When I look at `project_validator.go` file I'm confused. I can see a bunch of the test scenarios, but o description of what is going on. I need to switch between two files to understand the whole test scenario.

**Review by dharmit — COMMENTED** [review]
Commands for setup and execution are in the same function in most (all?) cases.

For example, we're creating the project and testing the actual thing in the same functions. Idea was to put setup related things in `JustBeforeEach` or something else like that and have only test specific code in the function.

**Pull request closed without merging**